File: Editorial/3047_Find_the_Largest_Area_of_Square_Inside_Two_Rectangles/Go/1_Approach_Traversal_Computational_Geometry.go
func largestSquareArea(bottomLeft [][]int, topRight [][]int) int64 {
    n := len(bottomLeft)
    var maxSide int64 = 0

    for i := 0; i < n; i++ {
        for j := i + 1; j < n; j++ {
            w := min(topRight[i][0], topRight[j][0]) - max(bottomLeft[i][0], bottomLeft[j][0])
            h := min(topRight[i][1], topRight[j][1]) - max(bottomLeft[i][1], bottomLeft[j][1])
            
            if w > 0 && h > 0 {
                side := min(w, h)
                if int64(side) > maxSide {
                    maxSide = int64(side)
                }
            }
        }
    }

    return maxSide * maxSide
}

File: Editorial/3001-3100/3047_Find_the_Largest_Area_of_Square_Inside_Two_Rectangles/C/1_Approach_Traversal_Computational_Geometry.c
long long largestSquareArea(int** bottomLeft, int bottomLeftSize,
                            int* bottomLeftColSize, int** topRight,
                            int topRightSize, int* topRightColSize) {
    int n = bottomLeftSize;
    long long maxSide = 0;

    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int w = fmin(topRight[i][0], topRight[j][0]) -
                    fmax(bottomLeft[i][0], bottomLeft[j][0]);
            int h = fmin(topRight[i][1], topRight[j][1]) -
                    fmax(bottomLeft[i][1], bottomLeft[j][1]);
            if (w > 0 && h > 0) {
                int side = fmin(w, h);
                if ((long long)side > maxSide) {
                    maxSide = (long long)side;
                }
            }
        }
    }

    return maxSide * maxSide;
}
